src/lib/rest/rest.cpp

```diff
@@ -1029,8 +1035,8 @@ static int connectionTreat
     // Add to the size of the accumulated read buffer
     ciP->payloadSize += *upload_data_size;
 
-    // Zero the payload (not really necessary)
-    ciP->payload[ciP->payloadSize + 1] = 0;
+    // Zero-terminate the payload
```

Comment:
Is this bugfix related with the topic of the PR? Or something that has been discovered during the process of implementing the MHD thread pool?